datastream-server-api/src/main/java/com/linkedin/datastream/server/api/transport/DatastreamRecordMetadata.java

```diff
@@ -71,9 +75,16 @@ public int getEventIndex() {
     return _eventIndex;
   }
 
+  /**
+   * Partition number from which the record was consumed.
+   */
+  public int getSourcePartition() {
+    return _sourcePartition;
+  }
+
   @Override
   public String toString() {
-    return String.format("Checkpoint: %s, Topic: %s, Partition: %d, Event Index: %d", _checkpoint, _topic, _partition,
-        _eventIndex);
+    return String.format("Checkpoint: %s, Topic: %s, Destination Partition: %d, Event Index: %d, Source Partition: %d",
```

Comment:
Nit: It may be worth renaming `partition` to `destinationPartition` to make it explicit to the users.

Reply:
Discussed offline and decided leave partition as is for now, since this is touched in many places and has some integration effort. Will open a ticket to fix this later.